packages/product/integration-tests/__tests__/services/product-variant/index.ts

```
@@ -251,4 +251,82 @@ describe("ProductVariant Service", () => {
       )
     })
   })
+
+  describe("retrieve", () => {
+    beforeEach(async () => {
+      testManager = await TestDatabase.forkManager()
+
+      productOne = testManager.create(Product, {
+        id: "product-1",
+        title: "product 1",
+        status: ProductTypes.ProductStatus.PUBLISHED,
+      })
+
+      variantOne = testManager.create(ProductVariant, {
+        id: "test-1",
+        title: "variant 1",
+        inventory_quantity: 10,
+        product: productOne,
+      })
+
+      await testManager.persistAndFlush([variantOne])
+    })
+
+    it("should return the requested variant", async () => {
+      const result = await service.retrieve(variantOne.id)
+
+      expect(result).toEqual(
+        expect.objectContaining({
+          id: "test-1",
+          title: "variant 1",
+        }),
+      )
+    })
+
+    it("should return requested attributes when requested through config", async () => {
+      const result = await service.retrieve(
+        variantOne.id,
+        {
+          select: ["id", "title", "product.title"] as any,
+          relations: ["product"],
+        }
+      )
+
+      expect(result).toEqual(
+        expect.objectContaining({
+          id: "test-1",
+          title: "variant 1",
+          product_id: "product-1",
+          product: expect.objectContaining({
+            id: "product-1",
+            title: "product 1",
+          }),
+        }),
+      )
+    })
+
+    it("should throw an error when a variant with ID does not exist", async () => {
+      let error
+
+      try {
+        await service.retrieve("does-not-exist")
+      } catch (e) {
+        error = e
+      }
+
+      expect(error.message).toEqual("ProductVariant with id: does-not-exist was not found")
+    })
+
+    it("should throw an error when a variant with ID does not exist", async () => {
```

todo: `should throw an error when not id is provided`?